### PR TITLE
handing in project 18

### PR DESCRIPTION
**Pull request by HenrikeW**
Merry Christmas and a Happy New Year!

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -12,27 +14,134 @@ import mongoose from 'mongoose'
...
+    const allTracks = await Track.find();
+    tracks = allTracks.filter((track) => track.genre.includes('pop'));
+  };
+
+  // get all tracks of an artist even when querying for parts of the name and ignoring upper/lower case
+  if (artistName) {
+    const allTracks = await Track.find();
+    tracks = allTracks.filter((track) => track.artistName.toLowerCase().includes(artistName.toLowerCase()));
+  };
+
```
Comment (sofiavazs) [inline server.js:103]: Maybe you can remove this line? ;)

File: server.js [hunk 2]
```diff
@@ -12,27 +14,134 @@ import mongoose from 'mongoose'
...
+// Get all tracks by a certain artist
+app.get('/artists/:id/tracks', async (req, res) => {
+  const artist = await Artist.findById(req.params.id);
+  if (artist) {
+    const tracks = await Track.find({ artistName: mongoose.Types.ObjectId(artist.id) });
+    res.json(tracks);
+  } else {
+    res.status(404).json({ error: 'Artist not found' })
+  };
+});
```
Comment (sofiavazs) [inline server.js:142]: 👏

**Review by sofiavazs — APPROVED** [review]
Great job Henrike! Your code is clean and easy to follow, super nice that you made two models too! 
I didn't have much to comment on the code itself but , for some reason when I was testing the endpoints I couldn't really get the error messages, even though you have error handling in your code, so maybe double check that? 
A suggestion for the readme, would have been nice to have the examples of endpoints written there as part of the documentation to help navigate through them, there are also some typos so maybe you want to correct them for the future ;) 

Merry Christmas and Happy New year! :D

**Review by KarolCislowski — APPROVED** [review]
Nice Project!

**Pull request closed without merging**